src/foundry/client/data/documents/chat-message.d.ts

```diff
@@ -60,132 +50,115 @@ declare global {
      */
     get isRoll(): boolean;
 
-    /**
-     * Return the Roll instance contained in this chat message, if one is present
-     */
-    get roll(): Roll | null;
-
     /**
      * Return whether the ChatMessage is visible to the current User.
      * Messages may not be visible if they are private whispers.
      */
     get visible(): boolean;
 
-    /**
-     * The User who created the chat message.
-     */
-    get user(): InstanceType<ConfiguredDocumentClass<typeof User>> | undefined;
-
-    override prepareData(): void;
+    prepareDerivedData(): void;
 
     /**
      * Transform a provided object of ChatMessage data by applying a certain rollMode to the data object.
-     * @param chatData - The object of ChatMessage data prior to applying a rollMode preference
-     * @param rollMode - The rollMode preference to apply to this message data
+     * @param chatData     - The object of ChatMessage data prior to applying a rollMode preference
+     * @param rollMode     - The rollMode preference to apply to this message data
      * @returns The modified ChatMessage data with rollMode preferences applied
      */
-    static applyRollMode(
-      chatData: ChatMessageDataConstructorData,
-      rollMode: keyof typeof CONFIG.Dice.rollModes | "roll"
-    ): ChatMessageDataConstructorData;
+    static applyRollMode(chatData: BaseChatMessage.ConstructorData, rollMode: string): ChatMessageData;
```

Comment:
The return value is probably also `BaseChatMessage.ConstructorData`.

Reply:
Updated